### DISCUSSION TITLE
Why GraphScope's interactive engine front end implemented in JAVA, what are the considerations?

### DISCUSSION THREAD

**Discussion by power1628** [alibaba/GraphScope#1783] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by welcome** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! And a maintainer will get back to you shortly!
Please feel free to contact us on [DingTalk](https://h5.dingtalk.com/circle/healthCheckin.html?dtaction=os&corpId=dingb4b76e0fa9ccd8d516fd0f0acdf19f55&68e71190=0fe34ec0&cbdbhh=qwertyuiop), WeChat account(graphscope) or [Slack](http://slack.graphscope.io/). We are happy to answer your questions responsively.

[the remark this answers is not in the document]
**Reply by sighingnow** [reply]
Could anybody @longbinlai @shirly121 give some insights on this topic?

[the remark this answers is not in the document]
**Reply by longbinlai** [reply]
The best reason is that FrontEnd always incorporates the compiler, and the compilers of widely-used graph query languages, such as Cypher and Gremlin, are in Java.
